### PR TITLE
Add AppDevAnalytics integration

### PR DESCRIPTION
**Pull request by cswenberg**
## Overview

- add [`AppDevAnalytics`](https://github.com/cuappdev/ios-analytics) cocoapod dependency (which depends on `Firebase`)
- add basic example of custom events and log placement for onboarding

## Changes Made

- update `Podfile` and `.lock` with Firebase dependencies
- add Firebase configuration file `GoogleService-Info.plist` and initialization in `AppDelegate.swift`
- new `Analytics/Analytics.swift` for defining custom event structs 
- new `Analytics/Events.md` for event documentation
- add basic example of custom events and log placement for onboarding
- debug flag option for Firebase to see live events come in on Firebase's [DebugView](https://console.firebase.google.com/u/3/project/volume-37b22/analytics/app/ios:com.cornellappdev.volume/debugview/~2F%3Ft%3D1614130747694&fpn%3D686490803994&swu%3D1&sgu%3D1&sus%3Dupgraded&cs%3Dapp.m.debugview.overview&g%3D1) (requires you to login with `analytics@cornellappdev.com` google credentials)
- my xcode autogenerated some details on my `schema.json` and `Secrets.plist` placement so can someone pls tell me how to fix it

# Test Coverage

- project successfully builds
- `AppDevAnalytics` automatically prints to console while on local dev -> read output
- turned on Firebase DebugView flag in a new `Build Scheme` and saw live events come in
- events should be fully logged into Firebase portal in <24hours to double-confirm logs made it to remote

## Next Steps (delete if not applicable)

- clean up xcode autogenerated file business
- update `AppDevAnalytics` repo link in Podfile once changes are merged from `conner/update` to `master`
- define more events to track as spec for another dev to implement
- push to Testflight to confirm end-to-end logging functionality
- confirm Firebase->BigQuery link working properly and set up BigQuery->Narrator link

## Related PRs or Issues (delete if not applicable)

- waiting on [ios-analytics](https://github.com/cuappdev/ios-analytics/pull/1) PR to merge

### CODE REVIEW

**Review by camruss — APPROVED** [review]
a-z 🔥

File: Volume/Supporting/AppDelegate.swift [hunk 1]
```diff
@@ -7,6 +7,7 @@
 //
 
 import UIKit
+import Firebase
```
Comment (danielVebman) [inline Volume/Supporting/AppDelegate.swift:10]: Alphabetize

File: Volume/Views/Onboarding/OnboardingView.swift [hunk 2]
```diff
@@ -7,6 +7,7 @@
 //
 
 import SwiftUI
+import AppDevAnalytics
```
Comment (danielVebman) [inline Volume/Views/Onboarding/OnboardingView.swift:10]: Alphabetize

File: Volume/Views/Onboarding/OnboardingView.swift [hunk 3]
```diff
@@ -87,6 +88,7 @@ struct OnboardingView: View {
                     Button("Start reading") {
                         withAnimation(.spring()) {
                             isFirstLaunch = false
+                            AppDevAnalytics.logEvent(CompleteOnboarding())
```
Comment (danielVebman) [inline Volume/Views/Onboarding/OnboardingView.swift:91]: Take this out of the `withAnimation` block (I'd put it right before on line 89). Same thing with the event on 121 below.

**Review by danielVebman — COMMENTED** [review]
`schema.json` goes in `Volume/Networking` and `Secrets.plist` goes in `Volume/Supporting`. Make sure to add `GoogleService-Info.plist` to `.gitignore`. I would place it in `Volume/Supporting` as well.

File: Volume/Views/Onboarding/OnboardingView.swift (hunk 3 above)
Comment (cswenberg) [reply Volume/Views/Onboarding/OnboardingView.swift:91]: thx I have no idea how swift UI codeblocks work 😅

File: Podfile [hunk 4]
```diff
@@ -8,5 +8,6 @@ target 'Volume' do
   # Pods for Volume
   pod 'Apollo'
   pod 'SDWebImageSwiftUI'
+  pod 'AppDevAnalytics', :git => 'https://github.com/cuappdev/analytics-ios.git', :commit => '5d459c0475'
```
Comment (danielVebman) [inline Podfile:11]: Why a specific commit?

File: README.md [hunk 5]
```diff
@@ -31,11 +31,18 @@ To access the project, clone the project, and run `pod install` in the project d
...
 
-3.  Finally, you need to generate `API.swift`. The Apollo iOS pod offers a build run script to do this. Go to the Volume target Build Phases and click the pluss to add a new Run Script Phase. Name it something like "Generate Apollo GraphQL API", and paste in the following script. When you build, Xcode will autogenerate `API.swift` if one of the query or mutation `.graphql` files has changed.
+AppDev members can access the `schema.json` file in the `#volume-dev` channel.
+
+3. AppDev uses Firebase for event logging which requires a `GoogleService-Info.plist` file in the project.
+
+AppDev members can access the `GoogleService-Info.plist` file via a pinned message in the `#volume-ios` channel.
+Place the file in the `Volume/` directory.
+
+4.  Finally, you need to generate `API.swift`. The Apollo iOS pod offers a build run script to do this. Go to the Volume target Build Phases and click the pluss to add a new Run Script Phase. Name it something like "Generate Apollo GraphQL API", and paste in the following script. When you build, Xcode will autogenerate `API.swift` if one of the query or mutation `.graphql` files has changed.
```
Comment (danielVebman) [inline README.md:45]: `pluss -> plus`

**Review by danielVebman — APPROVED** [review]
Noice

File: Podfile (hunk 4 above)
Comment (cswenberg) [reply Podfile:11]: I'm worried about needing to make breaking changes later in the semester. I have a few potential ideas to bump up the quality of the cocoapod, but need to experiment with a few apps first to see if its worth putting in the main pod.

**Pull request merged**